src/libsyntax/parse/lexer/unicode_chars.rs

```diff
@@ -0,0 +1,178 @@
+    ('〕', "Right Tortoise Shell Bracket", ')'),
+    ('﴿', "Ornate Right Parenthesis", ')'),
+    ('❴', "Medium Left Curly Bracket Ornament", '{'),
+    ('❵', "Medium Right Curly Bracket Ornament", '}'),
+    ('⁎', "Low Asterisk", '*'),
+    ('٭', "Arabic Five Pointed Star", '*'),
+    ('∗', "Asterisk Operator", '*'),
+    ('᜵', "Philippine Single Punctuation", '/'),
+    ('⁁', "Caret Insertion Point", '/'),
+    ('∕', "Division Slash", '/'),
+    ('⁄', "Fraction Slash", '/'),
+    ('╱', "Box Drawings Light Diagonal Upper Right To Lower Left", '/'),
+    ('⟋', "Mathematical Rising Diagonal", '/'),
+    ('⧸', "Big Solidus", '/'),
+    ('㇓', "Cjk Stroke Sp", '/'),
+    ('〳', "Vertical Kana Repeat Mark Upper Half", '/'),
+    ('丿', "Cjk Unified Ideograph-4E3F", '/'),
+    ('⼃', "Kangxi Radical Slash", '/'),
+    ('＼', "Fullwidth Reverse Solidus", '\\'),
+    ('﹨', "Small Reverse Solidus", '\\'),
+    ('∖', "Set Minus", '\\'),
+    ('⟍', "Mathematical Falling Diagonal", '\\'),
+    ('⧵', "Reverse Solidus Operator", '\\'),
+    ('⧹', "Big Reverse Solidus", '\\'),
+    ('㇔', "Cjk Stroke D", '\\'),
+    ('丶', "Cjk Unified Ideograph-4E36", '\\'),
+    ('⼂', "Kangxi Radical Dot", '\\'),
+    ('ꝸ', "Latin Small Letter Um", '&'),
+    ('﬩', "Hebrew Letter Alternative Plus Sign", '+'),
+    ('‹', "Single Left-Pointing Angle Quotation Mark", '<'),
+    ('❮', "Heavy Left-Pointing Angle Quotation Mark Ornament", '<'),
+    ('˂', "Modifier Letter Left Arrowhead", '<'),
+    ('꓿', "Lisu Punctuation Full Stop", '='),
+    ('›', "Single Right-Pointing Angle Quotation Mark", '>'),
+    ('❯', "Heavy Right-Pointing Angle Quotation Mark Ornament", '>'),
+    ('˃', "Modifier Letter Right Arrowhead", '>'),
+    ('Ⲻ', "Coptic Capital Letter Dialect-P Ni", '-'),
+    ('Ɂ', "Latin Capital Letter Glottal Stop", '?'),
+    ('Ⳇ', "Coptic Capital Letter Old Coptic Esh", '/'), ];
+
+const ASCII_ARRAY: &'static [(char, &'static str)] = &[
+    ('_', "Underscore"),
+    ('-', "Minus/Hyphen"),
+    (',', "Comma"),
+    (';', "Semicolon"),
+    (':', "Colon"),
+    ('!', "Exclamation Mark"),
+    ('?', "Question Mark"),
+    ('.', "Period"),
+    ('\'', "Quote"),
+    ('(', "Left Parenthesis"),
+    (')', "Right Parenthesis"),
+    ('{', "Left Curly Brace"),
+    ('}', "Right Curly Brace"),
+    ('*', "Asterisk"),
+    ('/', "Slash"),
+    ('\\', "Back Slash"),
+    ('&', "Ampersand"),
+    ('+', "Plus Sign"),
+    ('<', "Less-Than Sign"),
+    ('=', "Equals Sign"),
+    ('>', "Greater-Than Sign"), ];
+
+pub fn check_for_substitution(ch: char) -> Option<String> {
+    UNICODE_ARRAY
+    .iter()
+    .position(|&(c, _, _)| c == ch)
+    .map(|idx| {
+        let (_, u_name, ascii_char) = UNICODE_ARRAY[idx];
+        match ASCII_ARRAY.iter().position(|&(c, _)| c == ascii_char) {
+            Some(i) => {
+                let (_, ascii_name) = ASCII_ARRAY[i];
+                format!("Unicode character '{}' ({}) looks much like '{}' ({}), but it's not.",
```

Comment:
Do this after adding the test so that you don't have to rebuild, but this error message should start lowercase and not have a period at the end. (be sure to update the test with the error message)